src/H5Tconv.h

```diff
@@ -25,6 +25,15 @@
 /* Length of debugging name buffer */
 #define H5T_NAMELEN 32
 
+/* Swap two elements (I & J) of an array using a temporary variable */
+#define H5_SWAP_BYTES(ARRAY, I, J)                                                                           \
```

Comment:
Moved this macro out of H5Tconv.c to a more general place so it can be reused to make some of the conversion functions more compact.